frontend/server/src/DAO/Courses.php

```diff
@@ -601,6 +601,243 @@ public static function getStudentsProgressPerAssignment(
         ];
     }
 
+    /**
+     * Returns the list of assignments with their problems and points.
+     *
+     * @return array{assignmentsProblems: list<array{alias: string, name: string, points: float, problems: list<array{alias: string, title: string, isExtraProblem: bool, order: int, points: float}>, order: int}>, studentsProgress: list<array{assignments: array<string, array{problems: array<string, array{progress: float, score: float}>, progress: float, score: float}>, classname: string, country_id: null|string, globalProgress: float, globalScore: float, name: null|string, username: string}>}
+     */
+    public static function getStudentsProgressPerAssignmentv2(
+        int $courseId,
+        int $groupId,
+        int $page,
+        int $rowsPerPage
+    ): array {
+        $offset = ($page - 1) * $rowsPerPage;
+
+        $sqlAssignmentsProblems = '
+            SELECT
+                a.alias AS assignment_alias,
+                a.name AS assignment_name,
+                a.`order` AS assignment_order,
+                p.title AS problem_title,
+                p.alias AS problem_alias,
+                psp.is_extra_problem,
+                psp.points AS problem_points,
+                psp.`order` AS problem_order
+            FROM
+                Assignments a
+            INNER JOIN
+                Problemsets ps ON a.problemset_id = ps.problemset_id
+            INNER JOIN
+                Problemset_Problems psp ON psp.problemset_id = ps.problemset_id
+            INNER JOIN
+                Problems p ON p.problem_id = psp.problem_id
+            WHERE
+                a.course_id = ?
+            ORDER BY
+                a.`order`, psp.`order`';
+
+        $globalPoints = 0.0;
+        $assignmentsProblems = [];
+        /** @var list<array{assignment_alias: string, assignment_name: string, assignment_order: int, is_extra_problem: bool, problem_alias: string, problem_order: int, problem_points: float, problem_title: string}> */
+        $rs = \OmegaUp\MySQLConnection::getInstance()->GetAll(
+            $sqlAssignmentsProblems,
+            [ $courseId ]
+        );
+        foreach ($rs as $row) {
+            $globalPoints += $row['is_extra_problem'] ? 0.0 : $row['problem_points'];
+
+            if (!isset($assignmentsProblems[$row['assignment_alias']])) {
+                $assignmentsProblems[$row['assignment_alias']] = [
+                    'alias' => $row['assignment_alias'],
+                    'name' => $row['assignment_name'],
+                    'points' => 0.0,
+                    'problems' => [],
+                    'order' => $row['assignment_order'],
+                ];
+            }
+            $assignmentsProblems[$row['assignment_alias']]['points'] += $row['is_extra_problem'] ? 0.0 : $row['problem_points'];
```

Comment:
```suggestion
            if (!$row['is_extra_problem']) {
                $assignmentsProblems[$row['assignment_alias']]['points'] += $row['problem_points'];
                $coursePoints += $row['problem_points'];
            }
```

así hay menos ternarios!